### ISSUE TITLE
[BERT/PyTorch] question about distributed training 

### ISSUE THREAD

**Issue by vycezhong** [NVIDIA/DeepLearningExamples#785]
Hello. I have a question about distributed training of BERT. 

In [run_pretraining.py](https://github.com/NVIDIA/DeepLearningExamples/blob/master/PyTorch/LanguageModeling/BERT/run_pretraining.py) script, when `args.allreduce_post_accumulation` is `true`, it seems that allreduce does not start until all gradients are ready. Therefore, there is no overlap between computation and communication. Is my understanding correct? 

https://github.com/NVIDIA/DeepLearningExamples/blob/99b1c898cead5603c945721162270c2fe077b4a2/PyTorch/LanguageModeling/BERT/run_pretraining.py#L438-L451

**Comment by sharathts** [comment]
Yes. It only happens every args.gradient_accumulation_steps after all the gradients have been accumulated.

**Comment by vycezhong** [comment]
I see. Thanks for your explanation.
